.mvn/maven.config

```diff
@@ -1,2 +1,5 @@
+# Minimum Remoting version, which is tested for API compatibility
+# Specified here as a workaround so that renovate doesn't update it when it updates the main remoting dependency
+-Dremoting.minimum.supported.version=3107.v665000b_51092
```

Comment:
This workaround doesn't seem maintainable to me. I would be OK with less invasive workarounds, like not having a Maven property and duplicating the version number in the three places it's consumed (if that fixes the problem). Otherwise, download https://hub.docker.com/r/renovate/renovate, set [up logging](https://docs.renovatebot.com/config-overview/#logging-variables), run `renovate --platform=local`, and go to town debugging—it's all open-source JavaScript code, so it should be doable to add proper support for this use case, or at least to add enough print statements to find a cleaner workaround. :smile:

Reply:
why is it not maintainable to you? if someone is going to update the remoting minimum version it will be a conscious action and they will search the property find it and update.

Reply:
Using a system property rather than a Maven property unnecessarily increases visibility from the Maven API to the Java Platform API, and a wider surface area increases the likelihood something will be misused. In addition we've had problems in the past with IDEs not parsing `maven.config`; even if they've been resolved, this still seems like the wrong layer to define this. I've seen reviews on `apache/maven` where the Maven developers are pushing back against the use of the Java Platform APIs to get Maven properties, and there is a large category of Maven issues related to variable interpolation, so I've developed a gut feeling that a lot of subtle things can go wrong here with regard to Maven, hence my preference for a simpler workaround (such as duplicating the property, if that works) which might at least have a more understandable failure mode.